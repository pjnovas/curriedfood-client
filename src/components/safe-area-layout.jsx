import React from 'react';
import { View } from 'react-native';
import { useSafeArea } from 'react-native-safe-area-context';

export const SaveAreaInset = {
  TOP: 'top',
  BOTTOM: 'bottom'
};

export const SafeAreaLayout = (props) => {
  const safeAreaInsets = useSafeArea();
  const { insets, style, ...layoutProps } = props;

  const toStyleProp = (inset) => {
    switch (inset) {
      case SaveAreaInset.BOTTOM:
        return { paddingBottom: safeAreaInsets.bottom };
      case SaveAreaInset.TOP:
        return { paddingTop: safeAreaInsets.top };
    }
  };

  const createInsets = () => React.Children.map(insets, toStyleProp);

  return <View {...layoutProps} style={[style, createInsets()]} />;
};
