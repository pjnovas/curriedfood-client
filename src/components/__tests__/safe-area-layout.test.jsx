import React from 'react';
import { View } from 'react-native';
import { useSafeArea } from 'react-native-safe-area-context';
import { SafeAreaLayout, SaveAreaInset } from '../safe-area-layout';

const style = { flex: 1 };

jest.mock('react-native-safe-area-context');

useSafeArea.mockImplementation(() => ({
  top: 100,
  bottom: 50
}));

takeSnapshots(
  {
    default: {},
    withTop: {
      style,
      insets: SaveAreaInset.TOP,
      children: <View />
    },
    withBot: {
      style,
      insets: SaveAreaInset.BOTTOM,
      children: <View />
    }
  },
  SafeAreaLayout
);
