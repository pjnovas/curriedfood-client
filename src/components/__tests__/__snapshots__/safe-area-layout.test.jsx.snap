// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`default 1`] = `
<View
  style={
    Array [
      undefined,
      undefined,
    ]
  }
/>
`;

exports[`withBot 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
      },
      Array [
        Object {
          "paddingBottom": 50,
        },
      ],
    ]
  }
>
  <View />
</View>
`;

exports[`withTop 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
      },
      Array [
        Object {
          "paddingTop": 100,
        },
      ],
    ]
  }
>
  <View />
</View>
`;
